Android/Samples/libraries/kml/src/main/java/com/microsoft/maps/kml/KMLParser.java

```diff
@@ -79,46 +86,130 @@ private void parsePlacemark() throws IOException, XmlPullParserException, KMLPar
+  @NonNull
+  private MapElement parsePoint() throws IOException, XmlPullParserException, KMLParseException {
+    mParser.require(XmlPullParser.START_TAG, mNameSpace, "Point");
     while (mParser.next() != XmlPullParser.END_TAG) {
       if (mParser.getEventType() != XmlPullParser.START_TAG) {
         continue;
       }
       String type = mParser.getName();
-      if (type.equals("name")) {
-        mParser.require(XmlPullParser.START_TAG, mNameSpace, "name");
-        String title = parseText();
-        mParser.require(XmlPullParser.END_TAG, mNameSpace, "name");
-        return title;
+      if (type.equals("coordinates")) {
+        ArrayList<Geoposition> coordinates = parseCoordinates();
+        if (coordinates.size() > 1) {
+          throw new KMLParseException(
+              "coordinates for a Point can only contain one position. Instead saw: "
+                  + coordinates.size()
+                  + " at position: "
+                  + mParser.getPositionDescription());
+        }
+        MapIcon icon = mFactory.createMapIcon();
+        icon.setLocation(new Geopoint(coordinates.get(0)));
+        mParser.require(XmlPullParser.END_TAG, mNameSpace, "Point");
+        return icon;
       } else {
         skip();
       }
     }
-    return null;
+    // No coordintes element seen; throw error
+    throw new KMLParseException(
+        "<Point> must contain <coordinates> around XML position "
+            + mParser.getPositionDescription());
+  }
+
+  @NonNull
+  private ArrayList<Geoposition> parseCoordinates()
+      throws IOException, XmlPullParserException, KMLParseException {
+    mParser.require(XmlPullParser.START_TAG, mNameSpace, "coordinates");
+    String coordinates = parseText();
+    String[] allCoordinates = coordinates.split("\\s+");
+    ArrayList<Geoposition> positions = new ArrayList<>(allCoordinates.length);
+    for (String str : allCoordinates) {
+      String[] latLongAlt = str.split(",");
+      if (latLongAlt.length < 2) {
+        throw new KMLParseException(
+            "Error at: "
+                + mParser.getPositionDescription()
+                + " coordinates must contain at least latitude and longitude, separated by only a comma.");
+      }
+      double longitude = Double.parseDouble(latLongAlt[0]);
+      if (longitude < -180 || longitude > 180) {
+        throw new KMLParseException(
+            "Longitude must be in the range [-180, 180], instead saw: "
+                + longitude
+                + " at position: "
+                + mParser.getPositionDescription());
+      }
+      double latitude = Double.parseDouble(latLongAlt[1]);
+      if (latitude < -90 || latitude > 90) {
+        throw new KMLParseException(
+            "Latitude must be in the range [-90, 90], instead saw: "
+                + latitude
+                + " at position: "
+                + mParser.getPositionDescription());
+      }
+      double altitude = 0;
+      if (latLongAlt.length > 2) {
```

Comment:
What about the case where length is 4+? Do we silently ignore extra values that shouldn't be there?

Reply:
The GeoJSON spec explicitly states that more than 3 coordinates are allowed, but KML doesn't mention this. For now, I was taking the approach of silently ignoring the extra coordinates (which is what GeoJSON does). Would it be better to issue a warning or throw an error?

Reply:
Makes sense. I'm okay with keeping it as silently ignoring since GeoJSON does that too